src/tools/annotation/RectangleRoiTool.js

```diff
@@ -340,19 +346,44 @@ function _calculateStats(image, element, handles, modality, pixelSpacing) {
     (pixelSpacing.colPixelSpacing || 1) *
     (roiCoordinates.height * (pixelSpacing.rowPixelSpacing || 1));
 
-  const perimeter =
+  const perimeter = new Decimal(
     roiCoordinates.width * 2 * (pixelSpacing.colPixelSpacing || 1) +
-    roiCoordinates.height * 2 * (pixelSpacing.rowPixelSpacing || 1);
+      roiCoordinates.height * 2 * (pixelSpacing.rowPixelSpacing || 1)
+  );
+
+  const pixelDiagonal = measurementUncertainty.getPixelDiagonal(
+    pixelSpacing.colPixelSpacing,
+    pixelSpacing.rowPixelSpacing
+  );
+  const areaUncertainty = new Decimal(perimeter * pixelDiagonal);
+
+  const roundedArea =
+    measurementUncertainty.roundValueBasedOnUncertainty(
+      area,
+      areaUncertainty
+    ) || 0;
+
+  const roundedAreaUncertainty =
+    measurementUncertainty.roundValueBasedOnUncertainty(
+      areaUncertainty,
+      areaUncertainty
+    ) || 0;
+
+  const mean =
+    measurementUncertainty.getGenericRounding(roiMeanStdDev.mean) || 0;
+  const stdDev =
+    measurementUncertainty.getGenericRounding(roiMeanStdDev.stdDev) || 0;
```

Comment:
Same as in CircleRoiTool, I would suggest looking for a way to extract this logic to smaller functions so this big function is more readable and maintainable. Maybe we find logic that can be centralized and save some code duplication as well 👍